clients/venice-admin-tool/src/main/java/com/linkedin/venice/AdminTool.java

```diff
@@ -2810,6 +2821,40 @@ private static void cleanupInstanceCustomizedStates(CommandLine cmd) {
     printObject(multiStoreTopicsResponse);
   }
 
+  private static void getRequestBasedMetadata(CommandLine cmd) throws JsonProcessingException {
+    String url = getRequiredArgument(cmd, Arg.URL);
+    String storeName = getRequiredArgument(cmd, Arg.STORE);
+    ClientConfig clientConfig = ClientConfig.defaultGenericClientConfig(storeName).setVeniceURL(url);
+    if (clientConfig.isHttps()) {
+      if (!sslFactory.isPresent()) {
+        throw new VeniceException("HTTPS url requires admin tool to be executed with cert");
+      }
+      clientConfig.setSslFactory(sslFactory.get());
+    }
+    TransportClient transportClient = ClientFactory.getTransportClient(clientConfig);
+    getAndPrintRequestBasedMetadata(transportClient, url, storeName);
+  }
+
+  static void getAndPrintRequestBasedMetadata(TransportClient transportClient, String url, String storeName)
+      throws JsonProcessingException {
+    String requestBasedMetadataURL = QueryAction.METADATA.toString().toLowerCase() + "/" + storeName;
+    byte[] body;
+    try {
+      body = transportClient.get(requestBasedMetadataURL).get().getBody();
+    } catch (Exception e) {
+      throw new VeniceException(
+          "Encountered exception while trying to send metadata request to: " + url + "/" + requestBasedMetadataURL,
+          e);
+    }
+    RecordDeserializer<MetadataResponseRecord> metadataResponseDeserializer = FastSerializerDeserializerFactory
+        .getFastAvroSpecificDeserializer(MetadataResponseRecord.SCHEMA$, MetadataResponseRecord.class);
```

Comment:
Adding full compatibility (forward and backward) support to MetadataResponseRecord.

Reply:
Do we really need forward compat with admin topic? We can build and run this thing at any time. If needed I can write the support using controller client's schema fetch.

Reply:
Since the AdminTool won't operate against the Avro object, but just print it out, I guess we can just let Server return a json string instead of binary to make the E2E logic simpler, and also it will make it easier to the developer curl this endpoint directly without using AdminTool.
Transmitting binary in general is not good, but this is mainly for debug use, I guess it should be fine.

It is good to make sure AdminTool always work, otherwise, it will become troublesome during schema evolution since the user of AdminTool needs to build a version, which is using the same protocol version as backend.

Reply:
If we can use a Controller-based schema reader in the AdminTool, that will also work.
BTW, this metadata endpoint is also ACLed, right? If yes, that means the AdminTool needs to use a cert, which needs to be part of the read ACL list, which might be tedious.

Maybe we can exclude this metadata endpoint from the ACL enforcement,  plus the other debug endpoint: `AdminRequest`, and it can be done later.
Metadata doesn't contain any PII data, so I assume SSL should be good enough.